internal/adapters/redisadapters/redisadapters.go

```diff
@@ -0,0 +1,261 @@
+func (r *RedisAdapter) RemoveRefreshToken(ctx context.Context, refreshTokenID string) error {
+
+	return r.Rdb.Del(
+		ctx,
+		"refreshTokens-"+refreshTokenID,
+	).Err()
+}
+
+// RemoveFromIndexExpiringTokens removes an access token entry in the indexExpiringTokens sorted set from Redis
+func (r *RedisAdapter) RemoveFromIndexExpiringTokens(ctx context.Context, accessToken models.AccessToken) error {
+
+	var z1 redis.Z
+	z1.Score = float64(accessToken.ExpiresAt.Unix())
+	z1.Member = accessToken.ID
+
+	return r.Rdb.ZRem(
+		ctx,
+		"indexExpiringTokens",
+		z1,
+	).Err()
+}
+
+// RemoveProjectToken removes an access token entry in a projectTokens sorted set from Redis
+func (r *RedisAdapter) RemoveProjectToken(ctx context.Context, projectID int, accessToken models.AccessToken) error {
+
+	var z1 redis.Z
+	z1.Score = float64(accessToken.ExpiresAt.Unix())
+	z1.Member = accessToken.ID
+
+	return r.Rdb.ZRem(
+		ctx,
+		"projectTokens-"+strconv.Itoa(projectID),
+		z1,
+	).Err()
+}
+
+// Get functions
+
+// GetSession reads the associated ID, type, expiration and tokenID of a session from Redis
+func (r *RedisAdapter) GetSession(ctx context.Context, sessionID string) (models.Session, error) {
+
+	output, err := r.Rdb.HGetAll(
+		ctx,
+		"session-"+sessionID,
+	).Result()
+
+	expiresAtInt64, err := strconv.ParseInt(output["expiresAt"], 10, 64)
+
+	var accessTokenList []string
+	err = json.Unmarshal([]byte(output["tokenIds"]), &accessTokenList)
+
+	return models.Session{
+		ID:        sessionID,
+		Type:      output["type"],
+		ExpiresAt: time.Unix(expiresAtInt64, 0),
+		TokenIDs:  accessTokenList,
+	}, err
+}
+
+// GetAccessToken reads the associated ID, access token value, expiration, tokenID and refresh URL of an access token from Redis
+func (r *RedisAdapter) GetAccessToken(ctx context.Context, tokenID string) (models.AccessToken, error) {
+
+	output, err := r.Rdb.HGetAll(
+		ctx,
+		"accessTokens-"+tokenID,
+	).Result()
+
+	expiresAtInt64, err := strconv.ParseInt(output["expiresAt"], 10, 64)
+
+	return models.AccessToken{
+		ID:        tokenID,
+		Value:     output["accessToken"],
+		ExpiresAt: time.Unix(expiresAtInt64, 0),
+		URL:       output["URL"],
+		Type:      output["type"],
+	}, err
+}
+
+// GetRefreshToken reads the associated ID, refresh token value, expiration and tokenID of a refresh token from Redis
+func (r *RedisAdapter) GetRefreshToken(ctx context.Context, tokenID string) (models.RefreshToken, error) {
+
+	output, err := r.Rdb.HGetAll(
+		ctx,
+		"refreshTokens-"+tokenID,
+	).Result()
+
+	expiresAtInt64, err := strconv.ParseInt(output["expiresAt"], 10, 64)
+
+	return models.RefreshToken{
+		ID:        tokenID,
+		Value:     output["refreshToken"],
+		ExpiresAt: time.Unix(expiresAtInt64, 0),
+	}, err
+}
+
+// GetFromIndexExpiringTokens reads the associated expiration and tokenID of an access token from Redis
+func (r *RedisAdapter) GetFromIndexExpiringTokens(ctx context.Context, startTime int64, stopTime int64) ([]string, error) {
```

Comment:
Also it is best that the start and end time be `time.Time` types. Then inside the function we convert to unix timestamps. I also see that we already use `time.Time` in the structs for access/refresh tokens. So it makes sense to ask for `time.Time` in the signature here. It makes it more general IMO.

Reply:
My idea here was to keep the timestamps in int64, so that you could pass something like time.Now() + 3600 without having to then convert the result back into time.Time()

Reply:
You can do the same thing with time. And it is easier because you can add whole hours or different duration straight away without converting to seconds.

```go
time.Now().Add(time.Second * 3600)
time.Now().Add(time.Hour)
```

For comparison there is also `.Before` or `.After`.

There is also no confusion about what format/standard we use for the timestamps because `time.Time` is also timezone aware. So please change this to `time.Time`.